app/services/cycle_timetable.rb

```diff
@@ -14,6 +14,8 @@ def self.real_current_year
   end
 
   def self.current_year(now = Time.zone.now)
+    return if now.nil?
```

Comment:
😬

Reply:
I totally believe that this is necessary because we use this class all over and it's kind of weird, but would be good to know why, just to satisfy my curiosity.

Reply:
Should we keep it then?

Reply:
I think it's weird, so let's get rid of it unless we know it's going to break something otherwise. but it looks like you added it to handle where `application_choice_deferred_confirmed_at` was nil, so I would be in favour of not adding it, and just ensuring we don't pass nil to method.